src/dftbp/wavegrid/basis/lut.F90

```diff
@@ -0,0 +1,139 @@
+
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2025  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+!> Holds TRadialTableOrbital, a concrete implementation of TOrbital utilising a 1d interpolated lookup table.
+!> This allows for arbitrary radial functions.
+module dftbp_wavegrid_basis_lut
+  use dftbp_common_accuracy, only : dp
+  use dftbp_io_message, only : error
+  use dftbp_wavegrid_basis_orbital, only : TOrbital
+  implicit none
+
+  private
+
+  public :: TRadialTableOrbital, TRadialTableOrbital_initFromArray, TRadialTableOrbital_initFromOrbital
+  
+  !> Concrete class for an orbital represented by a linearly
+  !! interpolated lookup table.
+  type, extends(TOrbital) :: TRadialTableOrbital
+    !> Grid spacing (resolution)
+    real(dp) :: gridDist
+
+    !> Inverse of the grid spacing
+    real(dp) :: invLutStep
```

Comment:
if it is the inverse of `gridDist` it should be probably called `invGridDist` for consistency.